crates/core/src/types/engine/payload.rs

```diff
@@ -0,0 +1,158 @@
+    extra_data: Bytes,
+    #[serde(deserialize_with = "crate::serde_utils::u64::deser_hex_str")]
+    base_fee_per_gas: u64,
+    block_hash: H256,
+    transactions: Vec<EncodedTransaction>,
+    withdrawals: Vec<Withdrawal>,
+    #[serde(deserialize_with = "crate::serde_utils::u64::deser_hex_str")]
+    blob_gas_used: u64,
+    #[serde(deserialize_with = "crate::serde_utils::u64::deser_hex_str")]
+    excess_blob_gas: u64,
+}
+
+#[allow(unused)]
+#[derive(Debug)]
+pub struct EncodedTransaction(pub Bytes);
+
+impl<'de> Deserialize<'de> for EncodedTransaction {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Ok(EncodedTransaction(serde_utils::bytes::deser_hex_str(
+            deserializer,
+        )?))
+    }
+}
+
+#[allow(unused)]
+impl EncodedTransaction {
+    /// Based on [EIP-2718]
+    /// Transactions can be encoded in the following formats:
+    /// A) `TransactionType || Transaction` (Where Transaction type is an 8-bit number between 0 and 0x7f, and Transaction is an rlp encoded transaction of type TransactionType)
+    /// B) `LegacyTransaction` (An rlp encoded LegacyTransaction)
+    fn decode(&self) -> Result<Transaction, RLPDecodeError> {
+        // Look at the first byte to check if it corresponds to a TransactionType
+        match self.0.first() {
+            // First byte is a valid TransactionType
+            Some(tx_type) if *tx_type < 0x7f => {
+                // Decode tx based on type
+                let tx_bytes = &self.0.as_ref()[1..];
+                match *tx_type {
+                    // Legacy
+                    0x0 => {
+                        LegacyTransaction::decode_rlp(tx_bytes).map(Transaction::LegacyTransaction)
+                    } // TODO: check if this is a real case scenario
+                    // EIP1559
+                    0x2 => EIP1559Transaction::decode_rlp(tx_bytes)
+                        .map(Transaction::EIP1559Transaction),
+                    _ => unimplemented!("We don't know this tx type yet"),
+                }
+            }
+            // LegacyTransaction
+            _ => LegacyTransaction::decode_rlp(self.0.as_ref()).map(Transaction::LegacyTransaction),
+        }
+    }
+}
+
+// TODO: Consider adding a struct Block { block_header, block_body } and implementing TryInto
+#[allow(unused)]
+impl ExecutionPayloadV3 {
+    fn into_block(self) -> Result<(BlockHeader, BlockBody), RLPDecodeError> {
+        let block_body = BlockBody {
+            transactions: self
+                .transactions
+                .iter()
+                .map(|encoded_tx| encoded_tx.decode())
+                .collect::<Result<Vec<_>, RLPDecodeError>>()?,
+            ommers: vec![],
+            withdrawals: self.withdrawals,
+        };
+        Ok((
+            BlockHeader {
+                parent_hash: self.parent_hash,
+                ommers_hash: H256::zero(), // TODO: Use appropiate value
```

Comment:
We should use the keccak hash of an empty list: "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347". See https://eips.ethereum.org/EIPS/eip-3675#block-structure